src/unix/signal.c

```diff
@@ -63,6 +74,22 @@ static void uv__signal_global_init(void) {
 }
 
 
+static void uv__signal_global_reinit(void) {
+  /* We can only use signal-safe functions here.
+   * That includes read/write and close, fortunately.
+   * We do all of this directly here instead of resetting
+   * uv__signal_global_init_guard because
+   * uv__signal_global_once_init is only called from uv_loop_init
+   * and this needs to function in existing loops.
+   */
+  uv__close(uv__signal_lock_pipefd[0]);
+  uv__signal_lock_pipefd[0] = -1;
+  uv__close(uv__signal_lock_pipefd[1]);
+  uv__signal_lock_pipefd[1] = -1;
+  uv__signal_global_init();
+}
```

Comment:
I can do that, but I don't see how that could happen. The guard for `uv__signal_global_init` is a `static pthread_once_t` installed by `pthred_once`...if it's called again after a fork, then there are major bugs in the pthread library and we've got bigger problems 😄